### DISCUSSION TITLE
On tcp proxy with http handler.

### DISCUSSION THREAD

**Discussion by Andrewmatilde** [hyperium/hyper#2584] [Q&A] [answered]
I am writing a tcp proxy which could handle http packet with hyper . 
But when it comes to some custom method or error injected in http packet , hyper will suddenly return 4xx packet with `on_parse_error` . 
I want my proxy with hyper could just pass through these packet. 
Is there any way to handle the problem without write my custom hyper?
Thanks for any helps.

**Comment by seanmonstar** [comment]
Could you explain more what you expect to happen? hyper tries to be an HTTP handler, so if you get a TCP connection that is not sending HTTP, it will reject with an error. What sorts of "custom method" or "error injected" are you expecting to use?

**Comment by Andrewmatilde** [comment, marked as answer]
Thanks for your reply. The user of my tcp proxy used their custmomed HTTP method like such as `LOCK` and `UNLOCK` ... And we have tried to inject some HTTP error like close socket after send all body and some parse error on headers. These packet will be rejected with an error, but we want to proxy the error packet to target server to check if the server will panic or has some other problem.

**Reply by seanmonstar** [reply]
Custom HTTP methods work just fine with hyper, they won't return an error. But sending weird headers or closing the socket will, of course. Do you really want an HTTP server? It sounds like you just want a TCP proxy, in which case you don't really need hyper. You can just use 2 `TcpStream`s and [`tokio::io::copy_bidirectional`](https://docs.rs/tokio/1.*/tokio/io/fn.copy_bidirectional.html).

**Reply by Andrewmatilde** [reply]
Thanks for your advice.
